app/routes/questionnaire.py

```diff
@@ -104,34 +111,68 @@ def get_questionnaire(schema, questionnaire_store):
     )
 
     if not router.can_access_hub():
-        redirect_location_url = router.get_first_incomplete_location_in_survey_url()
+        redirect_location_url = (
+            router.get_first_incomplete_location_in_questionnaire_url()
+        )
         return redirect(redirect_location_url)
 
     if request.method == "POST":
-        if router.is_survey_complete():
+        if router.is_questionnaire_complete:
             submission_handler = SubmissionHandler(
                 schema, questionnaire_store, router.full_routing_path()
             )
             submission_handler.submit_questionnaire()
             return redirect(url_for("post_submission.get_thank_you"))
-        return redirect(router.get_first_incomplete_location_in_survey_url())
-
-    language_code = get_session_store().session_data.language_code
+        return redirect(router.get_first_incomplete_location_in_questionnaire_url())
 
-    hub = HubContext(
-        language=language_code,
+    hub_context = HubContext(
+        language=flask_babel.get_locale().language,
         schema=schema,
         answer_store=questionnaire_store.answer_store,
         list_store=questionnaire_store.list_store,
         progress_store=questionnaire_store.progress_store,
         metadata=questionnaire_store.metadata,
     )
-
-    hub_context = hub.get_context(
-        router.is_survey_complete(), router.enabled_section_ids
+    context = hub_context(
+        survey_complete=router.is_questionnaire_complete,
+        enabled_section_ids=router.enabled_section_ids,
+    )
+    return render_template(
+        "hub",
+        content=context,
+        page_title=context["title"],
     )
 
-    return render_template("hub", content=hub_context, page_title=hub_context["title"])
+
+@questionnaire_blueprint.route("submit/", methods=["GET", "POST"])
+@with_questionnaire_store
+@with_schema
+def submit(
+    schema: QuestionnaireSchema, questionnaire_store: QuestionnaireStore
+) -> Union[Response, str]:
+    try:
+        submit_handler = SubmitHandler(
+            schema, questionnaire_store, flask_babel.get_locale().language
+        )
+    except InvalidLocationException:
+        raise NotFound
+
+    if not submit_handler.router.is_questionnaire_complete:
+        return redirect(
+            submit_handler.router.get_first_incomplete_location_in_questionnaire_url()
+        )
+
+    if request.method == "POST":
+        submit_handler.handle_post()
+        return redirect(url_for("post_submission.get_thank_you"))
+
+    context = submit_handler.get_context()
+    return render_template(
+        submit_handler.template,
+        content=context,
```

Comment:
This wasn't originally done because the `page_title` comes from the context object and I wanted to avoid making the call twice.
```py
page_title=context["title"]
```